pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1598,36 +1587,12 @@ func (s *AWSCloud) EnsureTCPLoadBalancer(name, region string, publicIP net.IP, p
 	}
 
 	// Create a security group for the load balancer
-	var securityGroupID string
-	{
-		sgName := "k8s-elb-" + name
-		sgDescription := "Security group for Kubernetes ELB " + name
-		securityGroupID, err = s.ensureSecurityGroup(sgName, sgDescription, vpcId)
-		if err != nil {
-			glog.Error("Error creating load balancer security group: ", err)
-			return nil, err
-		}
-
-		permissions := []*ec2.IpPermission{}
-		for _, port := range ports {
-			portInt64 := int64(port.Port)
-			protocol := strings.ToLower(string(port.Protocol))
-			sourceIp := "0.0.0.0/0"
-
-			permission := &ec2.IpPermission{}
-			permission.FromPort = &portInt64
-			permission.ToPort = &portInt64
-			permission.IpRanges = []*ec2.IpRange{{CidrIp: &sourceIp}}
-			permission.IpProtocol = &protocol
-
-			permissions = append(permissions, permission)
-		}
-		_, err = s.ensureSecurityGroupIngress(securityGroupID, permissions)
-		if err != nil {
-			return nil, err
-		}
+	securityGroupId, err := s.createSecurityGroupForLoadBalancer(name, vpcId, ports)
```

Comment:
I do wonder if it's worth extracting out this block as a function - I realize that EnsureTCPLoadBalancer is too long, but it is nice to see exactly the policy we're putting onto the security groups.  I'm fine either way...